Serialization/Editor/ArgsListEditor.cs

```diff
@@ -26,7 +41,20 @@ public static void Draw(ArgsList argsList, IList<string> argNames = null, Dictio
 				{
 					argValue = EditorGUILayoutHelpers.FieldByType(label, argValue, argType);
 				}
-				argsList[i] = argValue;
+
+				if (EditorGUI.EndChangeCheck())
+				{
+					Undo.RecordObject(parentObject,
+						"Set " + parentObject.name + " argsList[" + label + "] to " + argValue);
+					argsList[i] = argValue;
+
+					// Undo functionality works, and should also mark objects as dirty,
+					// however for scriptable objects it won't mark them as dirty and you can't save changes
+					if (parentObject is ScriptableObject)
+					{
+						EditorUtility.SetDirty(parentObject);
```

Comment:
Did you look into `SerializedObject.ApplyModifiedProperties` to keep undo functionality for scriptable objects too?

Reply:
I'm not sure I can do that without using serialized properties